pkg/testcase/clusterrestore.go

```diff
@@ -0,0 +1,327 @@
+package testcase
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rancher/distros-test-framework/pkg/aws"
+	"github.com/rancher/distros-test-framework/pkg/customflag"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+var awsConfig shared.AwsConfig
+
+func setConfigs() {
+	awsConfig = shared.AwsConfig{
+		AccessKeyID:     os.Getenv("AWS_ACCESS_KEY_ID"),
+		SecretAccessKey: os.Getenv("AWS_SECRET_ACCESS_KEY"),
+	}
+
+}
+
+func TestClusterRestoreS3(
+	cluster *shared.Cluster,
+	applyWorkload bool,
+	flags *customflag.FlagConfig,
+) {
+	setConfigs()
+
+	product := cluster.Config.Product
+	_, version, err := shared.Product()
```

Comment:
Wont it be easier to get the version from the env `version := os.Getenv(fmt.Sprintf("%v_version", product))`